component/http/cache.go

```diff
@@ -0,0 +1,423 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/crc32"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/beatlabs/patron/log"
+)
+
+// CacheHeader is an enum representing the header value
+type CacheHeader int
```

Comment:
It is probably not used outside the http package, at least it should not.
```suggestion
type cacheHeader int
```

Reply:
actually it s not used at all.
First , tried to put all the possible header values as an enum ... but turned out to be unneeded abstraction. So will remove this type.